Branch-SDK/src/main/java/io/branch/referral/ServerRequestLogout.java

```diff
@@ -1,100 +0,0 @@
-package io.branch.referral;
-
-import android.app.Application;
-import android.content.Context;
-
-import org.json.JSONException;
-import org.json.JSONObject;
-
-/**
- * * <p>
- * The server request for logging out  current user from Branch API. Handles request creation and execution.
- * </p>
- */
-class ServerRequestLogout extends ServerRequest {
-
-    private Branch.LogoutStatusListener callback_;
-
-    /**
-     * <p>Create an instance of {@link ServerRequestLogout} to signal when  different person is about to use the app. For example,
-     * if you allow users to log out and let their friend use the app, you should call this to notify Branch
-     * to create a new user for this device. This will clear the first and latest params, as a new session is created.</p>
-     *
-     * @param context  Current {@link Application} context
-     * @param callback An instance of {@link io.branch.referral.Branch.LogoutStatusListener} to callback with the logout operation status.
-     */
-    public ServerRequestLogout(Context context, Branch.LogoutStatusListener callback) {
-        super(context, Defines.RequestPath.Logout);
-        callback_ = callback;
-        JSONObject post = new JSONObject();
-        try {
-            post.put(Defines.Jsonkey.RandomizedBundleToken.getKey(), prefHelper_.getRandomizedBundleToken());
-            post.put(Defines.Jsonkey.RandomizedDeviceToken.getKey(), prefHelper_.getRandomizedDeviceToken());
-            post.put(Defines.Jsonkey.SessionID.getKey(), prefHelper_.getSessionID());
-            if (!prefHelper_.getLinkClickID().equals(PrefHelper.NO_STRING_VALUE)) {
-                post.put(Defines.Jsonkey.LinkClickID.getKey(), prefHelper_.getLinkClickID());
-            }
-            setPost(post);
-        } catch (JSONException ex) {
-            ex.printStackTrace();
-            constructError_ = true;
-        }
-    }
-
-    public ServerRequestLogout(Defines.RequestPath requestPath, JSONObject post, Context context) {
-        super(requestPath, post, context);
-    }
-
-    @Override
-    public void onRequestSucceeded(ServerResponse resp, Branch branch) {
-        try {
-            prefHelper_.setSessionID(resp.getObject().getString(Defines.Jsonkey.SessionID.getKey()));
```

Comment:
Do these values also need to be cleared in `public void logout(LogoutStatusListener callback)`?

Reply:
Among the values cleared( set to NO_STRING_VALUE), its decided not to clear any referrer related params like Install params, Session params.